src/EventListener/LegacyRequestValueSynchronizingListener.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2016 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Request\ValueAdapter;
+use Contao\Environment;
+use Contao\Input;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Sets the current request on the \Input and \Environment classes.
+ *
+ * @author Christian Schiffler <https://github.com/discordier>
+ *
+ * @deprecated Deprecated since Contao 4.3, to be removed in Contao 5.0.
+ *             Use the request or request stack instead.
+ */
+class LegacyRequestValueSynchronizingListener implements EventSubscriberInterface
```

Comment:
> Because it needs to handle two events and has to do almost the same.
> Changing it to EventListeners would require 3 classes (two handling the events and one abstract base for implementing the common setter code) in order to not trigger duplicate code checks on Scrutinizer etc.

Whoot, no? Just register the same class for two tags:

```
tags:
    - { name: kernel.event_listener, event: kernel.request, method: startRequest }
    - { name: kernel.event_listener, event: kernel.response, method: finishRequest }
```